libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -578,6 +616,71 @@ bool AP_BattMonitor::reset_remaining(uint16_t battery_mask, float percentage)
     return ret;
 }
 
+// Logs smart battery messages
+void AP_BattMonitor::log_smart_battery_info() const
+{    
+    for(uint8_t i = 0; i < _num_instances; i++) {
+        if (get_type(i) > AP_BattMonitor::Type::ANALOG_VOLTAGE_AND_CURRENT) {
+            log_smart_battery_info(i);
+        }
+    }
+}
+
+void AP_BattMonitor::log_smart_battery_info(uint8_t instance) const
+{
+    uint16_t cycles;
+    bool got_cycle_count = get_cycle_count(instance,cycles);
+
+    uint16_t design_capacity;
+    bool got_design_capacity = get_design_capacity(instance, design_capacity);
+
+    int32_t capacity_full = pack_capacity_mah(instance); //battery._params->_pack_capacity
+
+    if (capacity_full == 0) {
+        capacity_full = -1;
+    }
```

Comment:
This breaks the pattern where you're using `got_foo` and substituting in some other value in `Write` below.  Choose one :-)

Reply:
Fixed to match the others. (Even though it adds an extra assignment?? ;) )